packages/@aws-cdk/aws-appmesh/lib/virtual-listener.ts

```diff
@@ -0,0 +1,338 @@
+        protocol: Protocol.GRPC,
+      },
+      healthCheck: renderHealthCheck({
+        port: this.port,
+        protocol: Protocol.GRPC,
+      }, this.healthCheck),
+      timeout: this.timeout ? renderTimeout(this.timeout, Protocol.GRPC) : undefined,
+    };
+  }
+}
+
+/**
+ * Represents the properties required to define a TCP Listener for a VirtualNode.
+ */
+class TcpNodeListener extends VirtualNodeListener {
+  /**
+   * Port to listen for connections on
+   *
+   * @default - 8080
+   */
+  readonly port: number;
+
+  /**
+   * Health checking strategy upstream nodes should use when communicating with the listener
+   *
+   * @default - no healthcheck
+   */
+  readonly healthCheck?: HealthCheck;
+
+  /**
+   * Listener timeout property for TCP protocol
+   *
+   * @default - none
+   */
+  readonly timeout?: TcpTimeout;
+
+  constructor(props?: TcpNodeListenerProps) {
+    super();
+    const checkedProps = props ?? {};
+    this.port = checkedProps.port ? checkedProps.port : 8080;
+    this.healthCheck = checkedProps.healthCheck;
+    this.timeout = checkedProps.timeout;
+  }
+
+  /**
+   * Return Listener for TCP protocol when Listener is added to Virtual Node.
+  */
+  public bind(_scope: cdk.Construct): CfnVirtualNode.ListenerProperty {
+    return {
+      portMapping: {
+        port: this.port,
+        protocol: Protocol.TCP,
+      },
+      healthCheck: renderHealthCheck({
+        port: this.port,
+        protocol: Protocol.TCP,
+      }, this.healthCheck),
+      timeout: this.timeout ? renderTimeout(this.timeout, Protocol.TCP) : undefined,
+    };
+  }
+}
+
+/**
+ * Represents the properties needed to define a Listeners for a VirtualNode
+ */
+interface VirtualNodeListenerProps {
+  /**
+   * Port to listen for connections on
+   *
+   * @default - 8080
+   */
+  readonly port?: number
+
+  /**
+   * The health check information for the listener
+   *
+   * @default - no healthcheck
+   */
+  readonly healthCheck?: HealthCheck;
+}
+
+/**
+ * Represent the HTTP Node Listener prorperty
+ */
+export interface HttpNodeListenerProps extends VirtualNodeListenerProps {
+  /**
+   * Protocol
+   *
+   * @default - HTTP
+   */
+  readonly protocol?: Protocol;
+  /**
+   * Timeout for HTTP protocol
+   *
+   * @default - None
+   */
+  readonly timeout?: HttpTimeout;
+}
+
+/**
+ * Represent the GRPC Node Listener prorperty
+ */
+export interface GrpcNodeListenerProps extends VirtualNodeListenerProps {
+  /**
+   * Timeout for GRPC protocol
+   *
+   * @default - None
+   */
+  readonly timeout?: GrpcTimeout;
+}
+
+/**
+ * Represent the TCP Node Listener prorperty
+ */
+export interface TcpNodeListenerProps extends VirtualNodeListenerProps {
+  /**
+   * Timeout for TCP protocol
+   *
+   * @default - None
+   */
+  readonly timeout?: TcpTimeout;
+}
+
+/**
+ * Returns an HealthCheck for a VirtualNode
+ */
+function renderHealthCheck(pm: PortMapping, hc: HealthCheck | undefined): CfnVirtualNode.HealthCheckProperty | undefined {
```

Comment:
This can be made a protected function of `VirtualNodeListener` and then you can use the `protocol` and `port` of the class instead of having to pass in the `PortMapping` interface.